src/event.rs

```diff
@@ -0,0 +1,64 @@
+use std::fmt;
+use std::borrow::Cow;
+
+use AttributeSet;
+use escape::Escaped;
+
+#[derive(Clone, Debug)]
+pub enum Event<'a> {
+    StartTag {
+        name: Cow<'a, str>,
+        attrs: AttributeSet<'a>,
+        self_closing: bool,
+    },
+    EndTag {
+        name: Cow<'a, str>,
+    },
+    Text(Cow<'a, str>),
+    RawHtml(Cow<'a, str>),
+}
+
+impl<'a> Event<'a> {
+    pub fn close_self(self) -> Event<'a> {
```

Comment:
Everyone may not know about the "self-closing" terminology, and having "self" in a method name seems weird!

```rust
start_tag!("br").close_self();
start_tag!("br").closed(); // I think this is clear enough
```